### PR TITLE
More work on general graphics 

### PR DESCRIPTION
**Pull request by maxwellfundi**
@dannyparsons This is still underway. Will let you know when its done. thank s

### CODE REVIEW

**Comment by maxwellfundi** [comment]
@dannyparsons  these are changes we made with David and i think they could be merged before i finish up other bits. 
Thanks

File: instat/clsRFunction.vb [hunk 1]
```diff
@@ -218,4 +218,26 @@ Public Class RFunction
         bIsAssigned = False
     End Sub
 
+    Public Function Clone() As RFunction
+
+        Dim clsRFunction As New RFunction
+        Dim clsRParam As New RParameter
```
Comment (dannyparsons) [inline instat/clsRFunction.vb]: You should not have this as a "New" RParameter, this creates a new RParameter at this point, where as you only want this to be used in the loop so it should just be Dim here.

File: instat/clsROperator.vb [hunk 2]
```diff
@@ -296,5 +296,36 @@ Public Class ROperator
...
         RemoveAllAdditionalParameters()
-    End Sub
+    End Sub
+
+    Public Function Clone() As ROperator
+        Dim clsTempROperator As New ROperator
+        Dim clsLeftFunction As New RFunction
+        Dim clsLeftOperator As New ROperator
+        Dim clsLeftParameter As New RParameter
+        Dim clsAdditionalParams As New RParameter
```
Comment (dannyparsons) [inline instat/clsROperator.vb]: Again, same argument here.

File: instat/clsRParameter.vb [hunk 3]
```diff
@@ -58,4 +58,19 @@ Public Class RParameter
         End If
         Return strRet
     End Function
+
+    Public Function Clone() As RParameter
+        Dim clsTempRParam As New RParameter
+        Dim clsArgumentOperator As New ROperator
```
Comment (dannyparsons) [inline instat/clsRParameter.vb]: Can you explain what clsArgumentOperator is for in this function?

File: instat/dlgBoxPlot.vb [hunk 4]
```diff
@@ -134,4 +137,10 @@ Public Class dlgBoxplot
         End If
         TestOkEnabled()
     End Sub
+
+    Private Sub ucrSaveBoxplot_GraphNameChanged() Handles ucrSaveBoxplot.GraphNameChanged
+        ucrBase.clsRsyntax.SetAssignTo(ucrSaveBoxplot.strGraphName, strTempDataframe:=ucrSelectorBoxPlot.ucrAvailableDataFrames.cboAvailableDataFrames.Text, strTempGraph:=ucrSaveBoxplot.strGraphName)
+        ucrBase.clsRsyntax.bExcludeAssignedFunctionOutput = True
+    End Sub
```
Comment (dannyparsons) [inline instat/dlgBoxPlot.vb]: Test this with multiple data frames, I think you will find a bug here.

**Comment by dannyparsons** [comment]
@maxwellfundi We've accepted this and looking to tidy it up a bit first.

**Comment by maxwellfundi** [comment]
Okay, Thanks alot.
This will make sure we make lots of good progress.

Thanks

Regards
Maxwell Fundi
|| Phone :+254728508241||

**Pull request merged**